CMakeLists.txt

```diff
@@ -171,6 +177,26 @@ if (WIN32)
   if (NOT OE_BASH)
     message(FATAL_ERROR "Git Bash not found!")
   endif ()
+
+  # Looking for Perl and Dos2unix if BUILD_OPENSSL is ON.
+  if (BUILD_OPENSSL)
+    # OpenSSL relies on Perl to generate files (e.g., headers, assembly files, and test cases).
+    find_program(OE_PERL perl PATHS "C:/Program Files/Git/bin"
+                                    "${GIT_DIR}/../usr/bin" NO_DEFAULT_PATH)
+    if (NOT OE_PERL)
+      message(FATAL_ERROR "Perl not found!")
+    endif ()
+
+    # Dos2unix utility is used to convert CRLF to LF, which is required by some
+    # OpenSSL tests.
+    find_program(
+      OE_DOS2UNIX dos2unix PATHS "C:/Program Files/Git/bin"
```

Comment:
`dos2unix` comes with the git bash by default, which is already our prerequisite, so it does not introduce new dependency.

The reason for this route is that we cannot configure the OpenSSL repo via `.gitattribute` because it does not apply to submodule. Besides, there is no good way of doing that via cmake. Then I found that `dos2unix` is a general tool (come with git bash) for this purpose.

I don't really know how OpenSSL handle those tests because the root cause is how the parsing logic is implemented.

Reply:
Thanks for the explanation.
It is good that dos2unix is not a new dependency.
If we expect that `dos2unix` will be available as part of git bash then it makes sense to error out if it is not found.

Otherwise we cannot be sure if all tests have been run or not.

Reply:
Updated.